test/built-ins/Array/fromAsync/this-constructor-with-bad-length-setter.js

```diff
@@ -0,0 +1,34 @@
+// Copyright (C) 2023 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-array.fromasync
+description: >
+  Rejects the promise if setting the length fails on an instance of a custom
+  this-value
+info: |
+  3.j.ii.4.a. Perform ? Set(_A_, *"length"*, 𝔽(_k_), *true*).
+  ...
+  3.k.viii. Perform ? Set(_A_, *"length"*, 𝔽(_len_), *true*)
+includes: [asyncHelpers.js]
+flags: [async]
+features: [Array.fromAsync]
+---*/
+
+asyncTest(async function () {
+  class MyArray {
+    set length(v) {
+      assert.sameValue(v, 3, "setter should be called with length 3");
```

Comment:
Hm, if this fails, won't it also throw a `Test262Error`? It seems potentially confusing to use it for both assertion failures and testing control flow.

Reply:
Good point. There are other tests that already confirm the value with which the length property is set, so I'll just remove this line.